components/content-service/pkg/storage/gcloud.go

```diff
@@ -187,9 +188,46 @@ func (rs *DirectGCPStorage) download(ctx context.Context, destination string, bk
 	span.SetTag("gcsObj", obj)
 	defer tracing.FinishSpan(span, &err)
 
-	rc, _, err := rs.ObjectAccess(ctx, bkt, obj)
-	if rc == nil {
-		return false, err
+	backupDir, err := os.MkdirTemp("", "backup-")
+	if err != nil {
+		return true, err
+	}
+	defer os.RemoveAll(backupDir)
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+
+		sa := ""
+		if rs.GCPConfig.CredentialsFile != "" {
+			sa = fmt.Sprintf(`-o "Credentials:gs_service_key_file=%v"`, rs.GCPConfig.CredentialsFile)
+		}
+
+		args := fmt.Sprintf(`gsutil -q -m %v\
+		  -o "GSUtil:sliced_object_download_max_components=8" \
+		  -o "GSUtil:parallel_thread_count=1" \
+		  cp gs://%s %s`, sa, filepath.Join(bkt, obj), backupDir)
```

Comment:
I think we should set the `-P` option because of keeping the file attribute. WDYT?
https://cloud.google.com/storage/docs/gsutil/commands/cp
> -P
Enables POSIX attributes to be preserved when objects are copied. gsutil cp copies fields provided by stat. These fields are the user ID of the owner, the group ID of the owning group, the mode or permissions of the file, and the access and modification time of the file. For downloads, these attributes are only set if the source objects were uploaded with this flag enabled.
> On Windows, this flag only sets and restores access time and modification time. This is because Windows doesn't support POSIX uid/gid/mode.

Reply:
I don't see a reason for that. The flag is for the tar file generated by `ws-daemon`, not the content.